packages/@aws-cdk/aws-codedeploy/lib/deployment-group.ts

```diff
@@ -134,6 +134,33 @@ export class InstanceTagSet {
   }
 }
 
+/**
+ * The configuration for automatically rolling back deployments in a given Deployment Group.
+ */
+export interface IAutoRollbackConfig {
+  /**
+   * Whether to automatically roll back a deployment that fails.
+   *
+   * @default true
+   */
+  failedDeployment?: boolean;
+
+  /**
+   * Whether to automatically roll back a deployment that was manually stopped.
+   *
+   * @default true
+   */
+  stoppedDeployment?: boolean;
+
+  /**
+   * Whether to automatically roll back a deployment during which one of the configured
+   * CloudWatch alarms for this Deployment Group went off.
+   *
+   * @default true
+   */
+  deploymentInAlarm?: boolean;
```

Comment:
I am assuming this requires that alarms are associated with the deployment group. Is there any meaning for this without alarms? Should we validate?

Reply:
I like the idea. I had to change the order the PRs have been submitted in to merge the alarms one (#926) before this one, but thankfully that one was approved already.

I've rebased this PR on top of the alarms one, added validation, and changed the default to depend on whether any Alarms were provided.